resources/js/interfaces/legacy-match-game-json.ts

```diff
@@ -13,12 +13,15 @@ export default interface LegacyMatchGameJson {
   mode: Ruleset;
   mode_int: number;
   mods: string[]; // TODO: use ModJson
-  scores: ScoreJson[]; // TODO: use SoloScoreJson
+  scores: LegacyMatchScoreJson[]; // TODO: use SoloScoreJson
   scoring_type: LegacyMatchScoringType;
   start_time: string;
   team_type: LegacyMatchTeamType;
 }
 
+// used to ensure presence of match
+export type LegacyMatchScoreJson = ScoreJson & { match: Match };
```

Comment:
I think this should work better in term of automatically matching type

```suggestion
export type LegacyMatchScoreJson = ScoreJson & Require<Pick<ScoreJson, 'match'>>
```